### PR TITLE
DTY-27 Added portals

### PR DESCRIPTION
**Pull request by devbucket**

### CODE REVIEW

File: components/NavDrawer/index.jsx [hunk 1]
```diff
@@ -119,82 +122,173 @@ class NavDrawer extends Component {
...
-        </Item>
+        <Portal name={commonPortals.NAV_MENU_PRIVACY_BEFORE} />
+        <Portal name={commonPortals.NAV_MENU_PRIVACY} props={props}>
+          <Item href={`${PAGE_PATH}/privacy`} icon={SecurityIcon} close={this.handleClose}>
+            <I18n.Text string="navigation.privacy" />
+          </Item>
+        </Portal>
+        <Portal name={commonPortals.NAV_MENU_PRIVACY_AFTER} />
 
         {showReturnPolicy && (
```
Comment (richardgorman) [inline components/NavDrawer/index.jsx]: The before and after portals should not depend on the return policy being shown or not.
There may be a case where an extension needs to come before the return policy. If it's then disabled then this portal is closed.

File: components/NavDrawer/index.jsx [hunk 2]
```diff
@@ -119,82 +122,173 @@ class NavDrawer extends Component {
...
+        <Portal name={categoryPortals.NAV_MENU_CATEGORIES_BEFORE} />
+        <Portal name={categoryPortals.NAV_MENU_CATEGORIES} props={props}>
+          <Item href={CATEGORY_PATH} icon={ViewListIcon} close={this.handleClose}>
+            <I18n.Text string="navigation.categories" />
           </Item>
-        }
+        </Portal>
+        <Portal name={categoryPortals.NAV_MENU_CATEGORIES_AFTER} />
+
+        {appConfig.hasFavorites && (
```
Comment (richardgorman) [inline components/NavDrawer/index.jsx]: Same comment as with return policy

**Comment by devbucket** [comment]
Tests are failing because the dependencies are not correct. They are not deployed yet.

File: components/NavDrawer/index.jsx [hunk 3]
```diff
@@ -119,82 +123,217 @@ class NavDrawer extends Component {
...
+        </Portal>
+        <Portal name={commonPortals.NAV_MENU_PRIVACY_AFTER} props={props} />
 
+        {/* Return Policy */}
+        <Portal name={marketPortals.NAV_MENU_RETURN_POLICY_BEFORE} props={props} />
         {showReturnPolicy && (
-          <Item href={`${PAGE_PATH}/return_policy`} icon={DescriptionIcon} close={this.handleClose}>
-            <I18n.Text string="navigation.return_policy" />
-          </Item>
+          <Fragment>
```
Comment (richardgorman) [inline components/NavDrawer/index.jsx]: Unnecessary Fragment

File: components/NavDrawer/index.jsx [hunk 4]
```diff
@@ -119,82 +123,217 @@ class NavDrawer extends Component {
...
+            <I18n.Text string="navigation.categories" />
           </Item>
-        }
+        </Portal>
+        <Portal name={categoryPortals.NAV_MENU_CATEGORIES_AFTER} props={props} />
+
+        {/* Favorites */}
+        <Portal name={favoritesPortals.NAV_MENU_FAVORITES_BEFORE} props={props} />
+        {appConfig.hasFavorites && (
+          <Fragment>
```
Comment (richardgorman) [inline components/NavDrawer/index.jsx]: Unnecessary Fragment

File: components/NavDrawer/index.jsx [hunk 5]
```diff
@@ -119,82 +123,217 @@ class NavDrawer extends Component {
...
         {user && (
-          <Item onClick={logout} icon={LogoutIcon} close={this.handleClose}>
-            <I18n.Text string="navigation.logout" />
-          </Item>
+          <Fragment>
+            <Portal
+              name={commonPortals.NAV_MENU_LOGOUT_BEFORE}
+              props={{
+                ...props,
+                handleLogout: logout,
```
Comment (richardgorman) [inline components/NavDrawer/index.jsx]: I think we don't need to pass this prop.

**Pull request merged**